src/utilstrencodings.cpp

```diff
@@ -46,6 +46,19 @@ string SanitizeFilename(const string& str)
     return strResult;
 }
 
+std::string HexInt(uint32_t val)
```

Comment:
This function can be replaced with one using the C++11 standard library.  Interactive test here: http://cpp.sh/9c44m
```
#include <string>
#include <iostream>
#include <sstream>
#include <iomanip>

std::string HexInt(uint32_t val)
{
    std::stringstream ss;
    ss << std::setfill('0') << std::setw(sizeof(uint32_t)*2) << std::hex << val;
    return ss.str();
}

int main()
{
    uint32_t branchId = 0xdeadbeef;
    std::cout << HexInt(branchId) << std::endl;
}
```

Reply:
Ooh nice, I'll go do that.